### PR TITLE
feat: support retrieving token balance for any account

### PR DESCRIPTION
**Pull request by jjleng**
### Description
1. Existing methods are only able to get token balance for the creator account. Add methods to get token balance for any accounts.
2. Improve the token tests. Add assertions.

### Test Plan
yarn test

### CODE REVIEW

File: ecosystem/typescript/sdk/src/token_client.ts [hunk 1]
```diff
@@ -278,16 +278,40 @@ export class TokenClient {
    * ```
    */
   async getTokenBalance(creator: MaybeHexString, collectionName: string, tokenName: string): Promise<Types.Token> {
```
Comment (davidiw) [inline ecosystem/typescript/sdk/src/token_client.ts:280]: I'd argue we shouldn't really have this function. It is unclear why we would ever want a specific function to query the balance for the creators account.

**Review by davidiw — COMMENTED** [review]
looks good, but I'd like to be more drastic!

File: ecosystem/typescript/sdk/src/token_client.ts (hunk 1 above)
Comment (gregnazario) [reply ecosystem/typescript/sdk/src/token_client.ts:280]: Yeah I'd agree.  Why not just call the per account one then?

Comment (gregnazario) [reply ecosystem/typescript/sdk/src/token_client.ts:280]: Let's create a github issue for these major version breaking changes that we want to do.

**Review by gregnazario — APPROVED** [review]

File: ecosystem/typescript/sdk/src/token_client.ts (hunk 1 above)
Comment (jjleng) [reply ecosystem/typescript/sdk/src/token_client.ts:280]: @davidiw we shouldn't have this for sure. This is an old API. The reason I did not kill this is that I don't want to introduce breaking changes. We are going to make it away in the next major release.

**Pull request merged**